drivers/media/i2c/imx219.c

```diff
@@ -538,16 +547,23 @@ static int imx219_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
 {
 	struct imx219 *imx219 = to_imx219(sd);
 	struct v4l2_mbus_framefmt *try_fmt =
-		v4l2_subdev_get_try_format(sd, fh->pad, 0);
+		v4l2_subdev_get_try_format(sd, fh->pad, IMAGE_PAD);
 
 	mutex_lock(&imx219->mutex);
 
-	/* Initialize try_fmt */
+	/* Initialize try_fmt for the image pad */
 	try_fmt->width = supported_modes[0].width;
 	try_fmt->height = supported_modes[0].height;
 	try_fmt->code = imx219_get_format_code(imx219);
 	try_fmt->field = V4L2_FIELD_NONE;
 
+	/* Initialize try_fmt for the embedded metadata pad */
+	try_fmt = v4l2_subdev_get_try_format(sd, fh->pad, METADATA_PAD);
+	try_fmt->width = IMX219_EMBEDDED_LINE_WIDTH;
+	try_fmt->height = IMX219_NUM_EMBEDDED_LINES;
+	try_fmt->code = MEDIA_BUS_FMT_SENSOR_DATA;
+	try_fmt->field = V4L2_FIELD_NONE;
+
```

Comment:
So one v4l2_subdev_get_try_format is called with the mutex held, and one without. If the comment on the mutex is to be believed, then it's only needed for the call to imx219_get_format_code.

Reply:
To be safe, i've done both v4l2_subdev_get_try_format calls before the mutex is held.

Reply:
v4l2_subdev_get_try_format is only getting a pointer to a structure within v4l2_subdev_pad_config (which is fh->pad).

As fh has only just been created, there is no issue over holding the mutex. It was more that one was acquired with it held and one without.